homeassistant/components/mpd/media_player.py

```diff
@@ -265,16 +268,44 @@ def media_album_name(self):
     @property
     def media_image_hash(self):
         """Hash value for media image."""
-        if file := self._currentsong.get("file"):
-            return hashlib.sha256(file.encode("utf-8")).hexdigest()[:16]
-
-        return None
+        return self._media_image_hash
 
     async def async_get_media_image(self):
         """Fetch media image of current playing track."""
         if not (file := self._currentsong.get("file")):
             return None, None
+        response = await self._async_get_file_image_response(file)
+        if not response:
+            return None, None
+
+        image = bytes(response.get("binary"))
+        mime = response.get(
+            "type", "image/png"
+        )  # readpicture has type, albumart does not
+        return (image, mime)
+
+    async def _async_update_media_image_hash(self):
+        """Update the hash value for the media image."""
+        file = self._currentsong.get("file")
+
+        if file == self._media_image_file:
```

Comment:
We should guard against `None` value here for `file` as we do above.
```suggestion
        if file is None or file == self._media_image_file:
```

Reply:
I think this won't work because then we won't set the file and hash to `None` if there is no file. I'll fix this in a second